DSC/dsc.py

```diff
@@ -264,14 +270,27 @@ def install_dsc_packages():
         deb_install_pkg(omi_package_path + '.x64.deb', 'omi', omi_version_deb)
         deb_install_pkg(dsc_package_path + '.x64.deb', 'dsc', dsc_version_deb )
     elif distro_category == DistroCategory.redhat or distro_category == DistroCategory.suse:
-        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi-' + omi_version_rpm)
-        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc-' + dsc_version_rpm)
+        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi')
+        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc')
+
+def compare_package_version(system_package_version, package_name):
+    if package_name == 'omi':
+        version = re.match('(\d+).(\d+).(\d+)', system_package_version)
+        if version is not None and (int(version.group(1)) > omi_major_version) or (int(version.group(1)) == omi_major_version and int(version.group(2)) >= omi_minor_version) or (int(version.group(1)) == omi_major_version and int(version.group(2)) == omi_minor_version and int(version.group(3)) >= omi_build):
+            return 1
+    elif package_name == 'dsc':
+       	version = re.match('(\d+).(\d+).(\d+)', system_package_version)
+        if version is not None and (int(version.group(1)) > dsc_major_version) or (int(version.group(1)) == dsc_major_version and int(version.group(2)) >= dsc_minor_version) or (int(version.group(1)) == dsc_major_version and int(version.group(2)) == dsc_minor_version and int(version.group(3)) >= dsc_build):
+            return 1
+    return 0			
 
 def rpm_install_pkg(package_path, package_name):
```

Comment:
I would split this function (rpm_install_pkg) to two - validate_installed_pck and install_pck. That will  make this easy to test where validate will only check if the version is installed and return if it is. In case, the right version is not installed - code will install.